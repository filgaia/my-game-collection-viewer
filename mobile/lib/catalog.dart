import 'game.dart';

class Catalog {
  late List<Game> ps;
  late List<Game> xbox;
  late List<Game> ns;
  late List<Game> wl;
  late String dv;
}
