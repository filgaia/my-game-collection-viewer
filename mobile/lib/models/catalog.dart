import 'package:json_annotation/json_annotation.dart';

import 'game.dart';

part 'catalog.g.dart';

@JsonSerializable(explicitToJson: true)
class Catalog {
  List<Game> ps;
  List<Game> xbox;
  List<Game> ns;
  List<Game> wl;
  String dv;

  Catalog(
      {required this.ps,
      required this.xbox,
      required this.ns,
      required this.wl,
      this.dv = 'ps'});

  factory Catalog.fromJson(Map<String, dynamic> json) =>
      _$CatalogFromJson(json);

  Map<String, dynamic> toJson() => _$CatalogToJson(this);
}
