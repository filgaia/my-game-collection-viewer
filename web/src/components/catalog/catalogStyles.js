const catalogStyles = (theme) => ({
    heroUnit: {
        backgroundColor: theme.palette.background.paper,
    },
    heroContent: {
        maxWidth: 600,
        margin: '0 auto',
        padding: `${theme.spacing.unit * 8}px 0 ${theme.spacing.unit * 6}px`,
    },
    layout: {
        width: 'auto',
        marginLeft: theme.spacing.unit * 3,
        marginRight: theme.spacing.unit * 3,
        [theme.breakpoints.up(1100 + theme.spacing.unit * 3 * 2)]: {
            width: 1100,
            marginLeft: 'auto',
            marginRight: 'auto',
        },
    },
    cardGrid: {
        padding: `${theme.spacing.unit * 8}px 0`,
    },
    card: {
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
    },
    cardMedia: {
        paddingTop: '56.25%', // 16:9
    },
    cardContent: {
        flexGrow: 1,
    },
    loading: {
        marginTop: '10px'
    },
    footer: {
        backgroundColor: theme.palette.background.paper,
        padding: theme.spacing.unit * 6,
    },
});

export default catalogStyles;
