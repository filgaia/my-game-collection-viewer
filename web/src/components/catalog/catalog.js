// @vendors
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import get from 'lodash/get';
import LazyLoad from 'react-lazyload';
import InfiniteScroll from 'react-infinite-scroller';
// @material
import {
    CircularProgress,
    CssBaseline,
    Grid,
    Typography
} from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
// @components
import GameCard from './../gameCard/gameCard';
// @styles
import catalogStyles from './catalogStyles';
// @constants
import { FIRST_PAGE } from '../../constants/index';

class Catalog extends Component {

    buildItems() {
        const { gamesInformation, games, setLabelFilter } = this.props;

        return (
            games.map(card => (
                <Grid item key={`card-${get(card, 'id')}`} xs={12} sm={6} md={4} lg={3}>
                    <LazyLoad height={250}>
                        <GameCard
                            idGame={get(card, 'id')}
                            description={get(card, 'description_short', '')}
                            title={get(card, 'name', '')}
                            image={get(card, 'image_url_medium', '')}
                            plataform={get(card, 'platform_id', 0)}
                            labels={get(card, 'labels', [])}
                            gamesInformation={gamesInformation}
                            setLabelFilter={setLabelFilter}
                        />
                    </LazyLoad>
                </Grid>
            ))
        );
    }

    buildCatalog(items, loader) {
        const { classes, loadGames, hasMoreItems } = this.props;

        return (
            <main>
                <InfiniteScroll
                    pageStart={FIRST_PAGE}
                    loadMore={loadGames}
                    hasMore={hasMoreItems}
                    loader={loader}
                >
                    <div className={classNames(classes.layout, classes.cardGrid)}>
                        <Grid container spacing={40}>
                            {items}
                        </Grid>
                    </div>
                </InfiniteScroll>
            </main>
        );
    }

    buildLoader() {
        const { classes } = this.props;

        return (
            <div key={0} className={classes.loading}>
                <CircularProgress />
            </div>
        ); // Key to remove warning of infinite scroll
    }

    buildError() {
        return (
            <Typography variant="h2" align="center" color="error" gutterBottom>
                There was an error loading your file.
            </Typography>
        );
    }

    render() {
        const { gamesInformation } = this.props;
        const loading = gamesInformation.get('loading');
        const error = gamesInformation.get('error');
        let catalogContainer = null;

        const loader = this.buildLoader();
        const items = this.buildItems();
        const errorMessage = this.buildError();

        if (error) {
            catalogContainer = errorMessage;
        } else if (loading) {
            catalogContainer = loader;
        } else {
            catalogContainer = this.buildCatalog(items, loader);
        }

        return (
            <React.Fragment>
                <CssBaseline />
                {catalogContainer}
            </React.Fragment >
        );
    }
}

Catalog.propTypes = {
    classes: PropTypes.object.isRequired,
    gamesInformation: PropTypes.object.isRequired,
    loadGames: PropTypes.func.isRequired,
    hasMoreItems: PropTypes.bool.isRequired,
    games: PropTypes.oneOfType([
        PropTypes.object,
        PropTypes.array
    ]).isRequired,
    setLabelFilter: PropTypes.func.isRequired
};

export default withStyles(catalogStyles)(Catalog);
