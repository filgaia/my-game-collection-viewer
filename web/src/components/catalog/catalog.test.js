import React from 'react';
import renderer from 'react-test-renderer';
import { Catalog } from './catalog';

describe('<Catalog />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Catalog
        classes={{}}
        gamesInformation={{ get: jest.fn() }}
        loadGames={jest.fn()}
        setLabelFilter={jest.fn()}
        hasMoreItems={false}
        games={[]}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
