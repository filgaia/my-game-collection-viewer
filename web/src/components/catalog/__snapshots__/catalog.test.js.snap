// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Catalog /> renders correctly 1`] = `
<main>
  <div>
    <div
      className=""
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      />
    </div>
  </div>
</main>
`;
